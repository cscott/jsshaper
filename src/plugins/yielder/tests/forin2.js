var log = (typeof console !== "undefined") && console.log || print;
(typeof require !== 'undefined') && require('setup.js');

function foo() {
    for (var o in { foo: 'bar', baz: 'bat' }) {
        yield o;
        b: break b;
        continue;
    }
}
function bar() {
    foo: for (var o in { foo2: 'bar2', baz2: 'bat2' }) {
        yield o;
        for (;;) break;
        bar: for (;;)
            continue foo;
    }
}

for (var x in foo()) {
    log(x);
}

for (var x in foo()) {
    log(x);
    break;
}

for (var x in bar()) {
    log(x);
}
