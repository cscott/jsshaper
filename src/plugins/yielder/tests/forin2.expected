foo
baz
foo
foo2
baz2
